intel_extension_for_transformers/transformers/utils/utility.py

```diff
@@ -152,3 +144,182 @@ def get_example_inputs_for_trace(model, quantization_config=None, return_type="d
+        attention_mask = torch.ones(input_bs, input_len)
+        position_ids = torch.arange(input_len).repeat(input_bs, 1)
+        if model.config.model_type != "opt":
+            example_inputs = {
+                "input_ids": input_ids,
+                "past_key_values": tuple(past_key_values),
+                "position_ids": position_ids,
+                "attention_mask": attention_mask
+            }
+            example_inputs = (input_ids, attention_mask, position_ids, tuple(past_key_values))
+        else:
+            example_inputs = {
+                "input_ids": input_ids,
+                "past_key_values": tuple(past_key_values),
+                "attention_mask": attention_mask
+            }
+            example_inputs = (input_ids, attention_mask, tuple(past_key_values))
+        # do inference to check example_inputs correct.
+        # out = model(**example_inputs)
+        return example_inputs
+
+def get_example_inputs_for_chatglm(model, quantization_config=None, return_type="dict"):
+    """
+        Generate the example_input for tracing, support models load from AutoModelForCausalLM.
+
+    """
+    query = "我该怎么办?"
+    tokenizer = quantization_config.tokenizer
+    if hasattr(tokenizer, "build_chat_inputs"):
+        inputs = tokenizer.build_chat_inputs(query)
+        eos_token_id = [tokenizer.eos_token_id, tokenizer.get_command("<|user|>"),
+                        tokenizer.get_command("<|observation|>")]
+        inputs["eos_token_id"] = eos_token_id
+        input_ids = inputs["input_ids"]
+        input_bs, input_len = input_ids.shape
+        attention_mask = torch.ones(input_bs, input_len + 1)
+        attention_mask[:, 0] = 0
+        past_key_values = generate_dummy_past_key_values(input_bs, model)
+        position_ids = inputs["position_ids"]
+    elif hasattr(tokenizer, "build_prompt"):
+        prompt = tokenizer.build_prompt(query)
+        inputs = tokenizer([prompt], return_tensors="pt")
+        input_ids = inputs["input_ids"]
+        input_bs, input_len = input_ids.shape
+        attention_mask = torch.ones(input_bs, input_len + 1)
+        attention_mask[:, 0] = 0
+        past_key_values = generate_dummy_past_key_values(input_bs, model)
+        position_ids = inputs["position_ids"]
+    else:
+        inputs = tokenizer([query], return_tensors="pt")
+        input_ids = inputs["input_ids"]
+        input_bs, input_len = input_ids.shape
+        attention_mask = torch.ones(input_bs, input_len + 1)
+        attention_mask[:, 0] = 0
+        past_key_values = generate_dummy_past_key_values(input_bs, model)
+        position_ids = torch.vstack([torch.arange(input_len) for i in range(input_bs)])
+    example_inputs = {
+        "input_ids": input_ids,
+        "past_key_values": tuple(past_key_values),
+        "position_ids": position_ids,
+        "attention_mask": attention_mask
+    }
+    return example_inputs
+
+from optimum.intel.generation.modeling import TSModelForCausalLM
```

Comment:
Can we lazy-import `TSModelForCausalLM` somehow so that everything doesn't need to rely on optimum  unnecessarily just because they `from .utility import LazyImport`?

Reply:
Fixed in #744